Lib/slapdtest/_slapdtest.py

```diff
@@ -277,9 +277,6 @@ def _cleanup_rundir(self):
         """
         Recursively delete whole directory specified by `path'
         """
-        # cleanup_rundir() is called in atexit handler. Until Python 3.4,
```

Comment:
The code is there for a specific reason. The ``cleanup_rundir`` is designed to be executed in an ``atexit`` handler. By the time the handler runs, the rest of the world may already be gone. You must keep both imports as long as python-ldap supports Python < 3.4.

Reply:
I understand the motivation for why this was originally added.

Having understood it, I'm asking for a way to demonstrate its usefulness to myself. [In my testing](https://github.com/python-ldap/python-ldap/pull/200#discussion_r178464223), whether or not the atexit handler used the top level import or the local import never made a practical difference. Neither failed, raised an exception, nor produced a warning.  If this is really necessary or useful, it should be possible to demonstrate it as such. How?

Right now, my concern is that it was added to avoid a issue that doesn't actually occur in practice.

Reply:
It occurs in practice and I ran into the issue pretty often during testing.

The problem not easy to reproduce. The bug occurs only on Python 2 and requires a combination of uncollectable reference cycles and bad luck. The problem with atexit handlers is a well-known and understood issue. If you search the internet for "Python atexit NoneType has no attribute", you'll find a lot of mentions.

To avoid the issue, a Python 2 atexit handlers must be self-contained and are not allowed to depend on any global variables or state. This especially includes module state, because modules are cleared first.